go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -554,10 +556,44 @@ func (c *cnxn) SetOption(key, value string) error {
 		}
 		c.timeouts.updateTimeout = timeout
 	case adbc.OptionKeyAutoCommit:
-		return adbc.Error{
-			Msg:  "[Flight SQL] transactions not yet supported",
-			Code: adbc.StatusNotImplemented,
+		autocommit := true
+		switch value {
+		case adbc.OptionValueEnabled:
+		case adbc.OptionValueDisabled:
+			autocommit = false
+		default:
+			return adbc.Error{
+				Msg:  "[Flight SQL] invalid value for option " + key + ": " + value,
+				Code: adbc.StatusInvalidArgument,
+			}
 		}
+
+		if autocommit != c.autocommit {
+			// if we're *enabling* autocommit, then commit any existing
+			// transaction. If we're *disabling* auto commit, start a
+			// transaction and store it.
+			ctx := metadata.NewOutgoingContext(context.Background(), c.hdrs)
+			var err error
+			if autocommit && c.txn != nil {
+				if err = c.txn.Commit(ctx, c.timeouts); err != nil {
+					return adbc.Error{
+						Msg:  "[Flight SQL] failed to update autocommit: " + err.Error(),
+						Code: adbc.StatusIO,
+					}
```

Comment:
The C++ driver polls GetSqlInfo on startup (inside AdbcDatabase) and checks if the server supports transactions in the first place, if not, it just acts like before/as if transactions aren't supported (feel free to punt that to another PR though but I think we'll need it eventually)

Reply:
(The key thing being that the Python driver manager always tries to toggle it since DBAPI defaults to autocommit off, so if it gets an error besides NotImplemented it throws an error to the user)

Reply:
Gotcha, i'll update this here to check GetSqlInfo on startup and return NotImplemented if that's the case.

Reply:
updated